src/plugins/intel_gpu/src/kernel_selector/kernels/rms/rms_kernel_bfyx_opt.cpp

```diff
@@ -58,14 +75,12 @@ JitConstants RMSKernelBfyxOpt::GetJitConstants(const rms_params& params, Dispatc
         }
 
         const std::string lws_0 = "get_local_size(0)";
-        // It can be expected that the maximum possible itemsNum will not exceed 32
-        // Therefore, in dynamic shape, stack_size including additional buffer is set to 33
-        constexpr size_t stack_size = 33;
+        auto item_num_and_lws = get_item_num_and_lws(params, stoi(data_size));
         jit.AddConstants({
             MakeJitConstant("DATA_SIZE", data_size),
             MakeJitConstant("LWS", lws_0),
```

Comment:
Set this to item_num_and_lws.second, and use __attribute__((reqd_work_group_size(LWS, 1, 1))) in kernel in rms_gpu_bfyx_opt.cl. I am seeing even better improvements with A770